cryptography-core/src/cdd_claim/cdd_claim_data.rs

```diff
@@ -29,6 +34,59 @@ impl CddClaimData {
     }
 }
 
+impl Encode for CddClaimData {
+    #[inline]
+    fn size_hint(&self) -> usize {
+        SCALAR_SIZE + SCALAR_SIZE
+    }
+
+    fn encode_to<W: Output>(&self, dest: &mut W) {
+        self.investor_did.as_bytes().encode_to(dest);
+        self.investor_unique_id.as_bytes().encode_to(dest);
+    }
+}
+
+impl Decode for CddClaimData {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, CodecError> {
+        let investor_did = <[u8; SCALAR_SIZE]>::decode(input)?;
+        let investor_did = Scalar::from_bits(investor_did);
+
+        let investor_unique_id = <[u8; SCALAR_SIZE]>::decode(input)?;
+        let investor_unique_id = Scalar::from_bits(investor_unique_id);
+
+        Ok(CddClaimData {
+            investor_did,
+            investor_unique_id,
+        })
+    }
+}
+
+#[derive(Debug, Copy, Clone)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct CddId(pub RistrettoPoint);
+
+impl Encode for CddId {
+    #[inline]
+    fn size_hint(&self) -> usize {
+        RISTRETTO_POINT_SIZE
+    }
+
+    fn encode_to<W: Output>(&self, dest: &mut W) {
+        self.0.compress().as_bytes().encode_to(dest);
+    }
+}
+
+impl Decode for CddId {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, CodecError> {
+        let id = <[u8; RISTRETTO_POINT_SIZE]>::decode(input)?;
```

Comment:
There are still several instances of less readable typing like this one.
```suggestion
        let id: [u8; RISTRETTO_POINT_SIZE] = decode(input)?;
```

Reply:
because this doesn't work for arrays.